apps/backoffices/src/app/guards/role.guard.ts

```diff
@@ -0,0 +1,44 @@
+import { Token } from '@castcle-api/utils/commons';
+import { CastcleException } from '@castcle-api/utils/exception';
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  SetMetadata,
+} from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { Request } from 'express';
+import { Observable } from 'rxjs';
+import { RoleStaff } from '../models/authentication.dto';
+import { Staff } from '../schemas/staff.schema';
+
+@Injectable()
+export class RoleGuard implements CanActivate {
+  constructor(private readonly reflector: Reflector) {}
+
+  canActivate(
+    context: ExecutionContext,
+  ): boolean | Promise<boolean> | Observable<boolean> {
+    const request = context.switchToHttp().getRequest();
+    const roles = this.reflector.get<RoleStaff[]>(
+      'roles',
+      context.getHandler(),
+    );
+    if (roles && roles.length > 0) {
+      const account = Token.decodeToken<Staff>(getTokenFromRequest(request));
+      if (roles.includes(account.role as RoleStaff)) {
+        return true;
+      }
+    }
+    return false;
+  }
+}
+
+const getTokenFromRequest = (request: Request) => {
+  const token = request.headers.authorization?.split(' ')?.[1];
+  if (!token) throw new CastcleException('MISSING_AUTHORIZATION_HEADERS');
+
+  return token;
+};
+
+export const Roles = (roles: RoleStaff[]) => SetMetadata('roles', roles);
```

Comment:
```suggestion
export const RequiredPermissions = (...permissions: Permission[]) => SetMetadata('permission', permissions);
```

ปล. ใช้ `...` จะได้เรียก `@RequiredPermissions(A, B, C)` ได้เลย ไม่ต้องใส่ []